### PR TITLE
[DOP-21799] Refactor celery initialization

### PR DESCRIPTION
**Pull request by IlyasDevelopment**
## Change Summary

- Moved Celery instances to separate backend, worker and scheduler modules
- Removed unnecessary variable definitions from `docs/conf.py`

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [x] Tests pass on CI and coverage does not decrease
* [ ] Documentation reflects the changes where applicable
* [ ] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/syncmaster/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: syncmaster/backend/celery.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# SPDX-FileCopyrightText: 2023-2024 MTS PJSC
+# SPDX-License-Identifier: Apache-2.0
+from syncmaster.backend.settings import ServerAppSettings
+from syncmaster.worker import celery_factory
+
+app = celery_factory(ServerAppSettings())
```
Comment (dolfinus) [inline syncmaster/backend/celery.py:6]: We don't use `python -m celery -A syncmaster.backend.celery` here, so why this is global?

Comment (dolfinus) [reply syncmaster/backend/celery.py:6]: Could we add separated Celery factories for server & scheduler, so they won't import anything from `syncmaster.worker`?

Comment (IlyasDevelopment) [reply syncmaster/backend/celery.py:6]: > We don't use `python -m celery -A syncmaster.backend.celery` here, so why this is a global variable created during module initialization?

We could create Celery instance in application factory and pass it via `Depends()`, but we also have to mock Celery in unit tests somehow

Comment (dolfinus) [reply syncmaster/backend/celery.py:6]: You could use something like that:
```
@fixture
def celery() -> Celery: ...


@fixture
def test_app(celery) -> FastAPI:
  app = application_factory()
  app.dependency_overrides[Celery] = lambda: app
  return app
```

Comment (IlyasDevelopment) [reply syncmaster/backend/celery.py:6]: > Could we add separated Celery factories for server & scheduler, so they won't import anything from `syncmaster.worker`?

1. We will still have to use `imports=["syncmaster.worker.transfer"]` in each instance
2. Implementation of `task_cls` (like `WorkerTask`) will be needed as our current tasks rely on `engine` and `settings.encryption.crypto_key`

Comment (dolfinus) [reply syncmaster/backend/celery.py:6]: 1. Are those imported anyway or just when celery worker receives a task to execute?
2. This is required only on worker side, not server/scheduler

Comment (IlyasDevelopment) [reply syncmaster/backend/celery.py]: Found the reason why it didn't work, fixed

File: syncmaster/scheduler/celery.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# SPDX-FileCopyrightText: 2023-2024 MTS PJSC
+# SPDX-License-Identifier: Apache-2.0
+from syncmaster.scheduler import celery_factory
+from syncmaster.scheduler.settings import SchedulerAppSettings
+
+app = celery_factory(SchedulerAppSettings())
```
Comment (dolfinus) [inline syncmaster/scheduler/celery.py:6]: Could we create celery app in `TransferJobManager.__init__`, and then patch this object in tests?

Comment (IlyasDevelopment) [reply syncmaster/scheduler/celery.py:6]: The reason is that method `TransferJobManager.send_job_to_celery` should be static, therefore we can't pass celery app via self or as a parameter

File: syncmaster/worker/celery.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# SPDX-FileCopyrightText: 2023-2024 MTS PJSC
+# SPDX-License-Identifier: Apache-2.0
+from syncmaster.worker import celery_factory
+from syncmaster.worker.settings import WorkerAppSettings
+
+app = celery_factory(WorkerAppSettings())
```
Comment (dolfinus) [inline syncmaster/worker/celery.py:6]: Shouldn't we update `python -m celery -A syncmaster.worker.celery` entrypoint & usage examples to use a new location?

File: syncmaster/scheduler/celery.py (hunk 2 above)
Comment (IlyasDevelopment) [reply syncmaster/scheduler/celery.py:7]: Added explanation

File: syncmaster/worker/celery.py (hunk 3 above)
Comment (IlyasDevelopment) [reply syncmaster/worker/celery.py:6]: The path remains the same, Celery supports it https://docs.celeryq.dev/en/stable/getting-started/next-steps.html#about-the-app-argument

**Review by dolfinus — APPROVED** [review]

**Pull request merged**